### DISCUSSION TITLE
nvs_read return value for entries with data of length 0

### DISCUSSION THREAD

**Discussion by r2r0** [zephyrproject-rtos/zephyr#63843] [Q&A] [answered]
Hi,

nvs_read returns ENONT error for entries with data length equal to 0.
It make impossible to distinguish between deleted/non present entries and entries with trimmed data.
Is it correct behavior?

Regards,

**Comment by Laczen** [comment, marked as answer]
Hi @r2r0, doing a write with length equal to zero is equivalent to a delete. This is used as an indication that this data does not need to be kept. When trying to read a deleted item it returns -ENOENT.

**Reply by r2r0** [reply]
Hi @Laczen, tanks for answer.
What about making this behaior explicite in documentation?
If needed I can make small PR.

**Reply by Laczen** [reply]
A PR to update the documentation would be welcome, thanks.
